src/main/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseServiceImpl.java

```diff
@@ -202,11 +200,43 @@ public CaseEvent createCaseEvent(CaseEvent caseEvent, Case newCase, Timestamp ti
         // events which do not cause CaseGroupStatus transitions, (this is valid behaviour).
         log.debug(e.getMessage());
       }
+
+      // if this is a respondent enrolling event
+      if (caseEvent.getCategory().toString().equals("RESPONDENT_ENROLED")) {
+        // are there other case groups that need updating
+        List<CaseGroup> caseGroups = caseGroupService.transitionOtherCaseGroups(category, caseEvent, targetCase, newCase);
+        checkCaseState(category, caseGroups, caseEvent, newCase);
+      } else {
+        // should a new case be created?
+        createNewCase(category, caseEvent, targetCase, newCase);
+        effectTargetCaseStateTransition(category, targetCase);
+      }
     }
 
     return createdCaseEvent;
   }
 
+  /**
+   * This has been triggered by a 'RESPONDENT_ENROLED' event. If we find any associated Case Groups
+   * with only B cases we need to make case 'INACTIONABLE' and create BI case.
+   */
+  private void checkCaseState(final Category category, List<CaseGroup> caseGroups, CaseEvent caseEvent, Case newCase) throws CTPException {
+    // check all case groups are type B. For surveys this is always true
+    List<CaseGroup> caseGroupsToUpdate = caseGroups.stream().filter(cg -> cg.getSampleUnitType().toString().equals("B")).collect(Collectors.toList());
+    for(CaseGroup cg : caseGroupsToUpdate) {
+      // fetch cases associated to case group
+      List<Case> cases = caseRepo.findByCaseGroupFKOrderByCreatedDateTimeDesc(cg.getCaseGroupPK());
+      // find b cases
+      List<Case> bCases = cases.stream().filter(c -> c.getSampleUnitType().toString().equals("B")).collect(Collectors.toList());
+      // see if any case needs to transition
+      for(Case bCase : bCases) {
+        effectTargetCaseStateTransition(category, bCase);
+        Case c = new Case();
```

Comment:
I would rename the case to biCase. I was confused until reading the comment for the method.